packages/chain/statemgr/syncmgr.go

```diff
@@ -90,7 +99,7 @@ func (sm *stateManager) doSyncActionIfNeeded() {
 			return
 		}
 		nowis := time.Now()
-		if nowis.After(requestBlockRetryTime) {
+		if approvedBlockCandidatesCount == 0 && nowis.After(requestBlockRetryTime) {
```

Comment:
Sometimes even when having plenty of blocks, other nodes must be requested.
```suggestion
		if !sm.syncingBlocks.isObtainedFromWall(i) && nowis.After(requestBlockRetryTime) {
```
`isObtainedFromWall` must also be implemented